### ISSUE TITLE
feat: Using Multi-auth for authentication plugins with type other than type 'auth'.

### ISSUE THREAD

**Issue by SatriyoJati** [apache/apisix#11599]
### Description

Currently , I want to integrate keycloak or 3rd party identity server with Apisix. So when we call the our apis protected by Apisix, its need to provide credentials (resulted in an access token ) that has been generated from keycloak. Then we also want to use key-auth plugin as well so our apis protected by apikey. We cannot use multi-auth for that . We need to provide both credentials when accessing apis.  What I want to use either apikey or token, so when we call using apikey we dont need to provide token, and vice versa.

**Comment by jinpy666** [comment]
Yes, I had the same problem. Eventually, I implemented a new auth plugin instead of the built-in one. This requires an external system to manage the connection authorization of multiple users/systems, only the consumer is insufficient.
